openedx/core/djangoapps/schedules/management/commands/send_recurring_nudge.py

```diff
@@ -0,0 +1,121 @@
+
+
+class ScheduleStartResolver(RecipientResolver):
+    def __init__(self, site, current_date):
+        self.site = site
+        self.current_date = current_date.replace(hour=0, minute=0, second=0)
+
+    def send(self, week):
+        if not ScheduleConfig.current(self.site).enqueue_recurring_nudge:
+            return
+
+        target_date = self.current_date - datetime.timedelta(days=week * 7)
+        for hour in range(24):
+            target_hour = target_date + datetime.timedelta(hours=hour)
+            _schedule_hour.apply_async((self.site.id, week, target_hour), retry=False)
+
+
+@task(ignore_result=True, routing_key=settings.ACE_ROUTING_KEY)
+def _schedule_hour(site_id, week, target_hour):
+    msg_type = RecurringNudge(week)
+
+    for (user, language, context) in _schedules_for_hour(target_hour):
+        msg = msg_type.personalize(
+            Recipient(
+                user.username,
+                user.email,
+            ),
+            language,
+            context,
+        )
+        _schedule_send.apply_async((site_id, msg), retry=False)
+
+
+@task(ignore_result=True, routing_key=settings.ACE_ROUTING_KEY)
+def _schedule_send(site_id, msg):
+    site = Site.objects.get(pk=site_id)
+    if not ScheduleConfig.current(site).deliver_recurring_nudge:
+        return
+
+    ace.send(msg)
+
+
+def _schedules_for_hour(target_hour):
+    schedules = Schedule.objects.select_related(
+        'enrollment__user__profile',
+        'enrollment__course',
+    ).filter(
+        start__gte=target_hour,
+        start__lt=target_hour + datetime.timedelta(minutes=60),
+    )
+
+    if "read_replica" in settings.DATABASES:
+        schedules = schedules.using("read_replica")
+
+    for schedule in schedules:
+        enrollment = schedule.enrollment
+        user = enrollment.user
+
+        course_id_str = str(enrollment.course_id)
+        course = enrollment.course
+
+        # TODO: this produces a URL that contains the literal "+" character in the course key, which breaks sailthru
+        course_root = reverse('course_root', kwargs={'course_id': course_id_str})
+
+        def absolute_url(relative_path):
+            return u'{}{}'.format(settings.LMS_ROOT_URL, relative_path)
+
+        template_context = {
+            'student_name': user.profile.name,
+            'course_name': course.display_name,
+            'course_url': absolute_url(course_root),
+
+            # This is used by the bulk email optout policy
+            'course_id': course_id_str,
+        }
+
+        yield (user, course.language, template_context)
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument('--date', default=datetime.datetime.utcnow().date().isoformat())
+        parser.add_argument('site_domain_name')
+
+    def handle(self, *args, **options):
+        current_date = datetime.datetime(
+            *[int(x) for x in options['date'].split('-')],
```

Comment:
I don't think you can force strptime to be a timezone aware UTC date, unfortunately.